lib/ASRouter.jsm

```diff
@@ -134,6 +150,21 @@ const MessageLoaderUtils = {
   REMOTE_LOADER_CACHE_KEY: "RemoteLoaderCache",
   _errors: [],
 
+  /**
+   * This array is used to avoid the duplicate Remote Settings attachment downloads
+   * for `loadMessagesFromAllProviders`. For each download request, it records the
+   * `record ID` of the corresponding attachment, so the subsequent downloads for
+   * the same record can be simply skipped.
+   *
+   * Once all the providers complete the message load, one should reset this array
+   * via `this.resetRemoteAttachmentFetched()`.
+   */
+  _remoteAttachmentFetched: [],
+
+  resetRemoteAttachmentFetched() {
+    this._remoteAttachmentFetched = [];
+  },
```

Comment:
I'm not sure this is necessary.

The attachment downloader is already a no-op if the file already exists locally. So basically this allows to save the HTTP requests that pull the record metadata. But if I'm not mistaken, those requests will be cached.

Reply:
> The attachment downloader is already a no-op if the file already exists locally.

I assume that downloader will also check the checksum against the remote apart from the file existence, correct?

If that's the case, yeah, this check is unnecessary then, I can drop it for sure.